bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/TypeCreator.java

```diff
@@ -279,7 +281,7 @@ public static StreamType createStreamType(Type constraint, Type completionType)
      */
     public static StreamType createStreamType(String typeName, Type constraint,
                                               Type completionType, Module modulePath) {
-        return new BStreamType(typeName, constraint, completionType, modulePath);
+        return new BStreamType(IdentifierUtils.decodeIdentifier(typeName), constraint, completionType, modulePath);
```

Comment:
Should we always expect decoded type name from api. Do we need this. In other places as well in TypeCreator?

Reply:
I think we need this decoding if we are to remove it in BType constructor.
Because, we use the decoded names in all places other than the definitions in generated classes. Eg. call methods/ switch cases etc.
So it may cause issues if we don't use the decoded names as type names at runtime.

Reply:
I have removed these